### PR TITLE
Fix ClassCastException inserting a paragraph into header or footer (bug 69042)

### PR DESCRIPTION
**Pull request by AntonOellerer**
Up until now, when trying to insert a paragraph into a header or footer via the `XWPFDocument.insertNewParagraph(XmlCursor)` method, a `ClassCastException` was thrown as the code
 incorrectly assumed that the object in the path to the cursor is a table in all cases.
To fix this, the method was updated so that the case where the path object is the header or footer is handled appropriately.
https://bz.apache.org/bugzilla/show_bug.cgi?id=69042

### CODE REVIEW

File: poi-ooxml/src/main/java/org/apache/poi/xwpf/usermodel/XWPFDocument.java [hunk 1]
```diff
@@ -912,6 +920,53 @@ private void insertElementIntoCell(XWPFTableCell tableCell, IBodyElement iBodyEl
         }
     }
 
+    private void insertIntoTables(IBodyElement iBodyElement, Deque<XmlObject> path, CTTbl ctTbl, List<XWPFTable> tables) {
+        for (XWPFTable xwpfTable : tables) {
+            if (ctTbl == xwpfTable.getCTTbl()) {
+                insertElementIntoTable(xwpfTable, iBodyElement, path);
+            }
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/xwpf/usermodel/XWPFDocument.java:930]: shouldn't you break when you match in the `if`?

File: poi-ooxml/src/main/java/org/apache/poi/xwpf/usermodel/XWPFDocument.java [hunk 2]
```diff
@@ -912,6 +920,53 @@ private void insertElementIntoCell(XWPFTableCell tableCell, IBodyElement iBodyEl
...
+        }
+    }
+
+    private void insertIntoHeadersFooters(IBodyElement iBodyElement, CTHdrFtrImpl ctHdrFtrImpl) {
+        for (XWPFHeader header : headers) {
+            if (ctHdrFtrImpl == header._getHdrFtr()) {
+                insertIntoHeaderFooter(iBodyElement, header);
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/xwpf/usermodel/XWPFDocument.java:937]: same issue as above about break call when you match in the if and likewise in next for loop.

File: poi-ooxml/src/test/java/org/apache/poi/xwpf/TestXWPFBugs.java [hunk 3]
```diff
@@ -25,9 +25,7 @@ Licensed to the Apache Software Foundation (ASF) under one or more
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
-
 import javax.crypto.Cipher;
-
```
Comment (pjfanning) [inline poi-ooxml/src/test/java/org/apache/poi/xwpf/TestXWPFBugs.java]: Can you add back this new line?

**Comment by AntonOellerer** [comment]
Thank you for the review, I added `breaks`/`returns` where necessary!

**Comment by AntonOellerer** [comment]
I did some more work around this problem and discovered that the problem might actually lie somewhere else, as `XWPFDocument.insertNewParagraph(XmlCursor)` should not be invoked if the parent object is actually a `XWPFHeaderFooter`, instead we should use `XWPFHeaderFooter.insertNewParagraph(XmlCursor)`.
What is your opinion on this @pjfanning?

**Comment by pjfanning** [comment]
My preference is to make as few changes as possible to POI. It is a 20 year old project and it is very easy to break stuff when attempting to make changes.

**Pull request closed without merging**